### PR TITLE
Restore Julia CI

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: .github/workflows/julia-auto.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: hdf5 1.14 autotools julia
```
Comment (byrnHDF) [inline .github/workflows/julia-auto.yml]: 1.14 to dev

File: .github/workflows/julia-cmake.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Sources
+        uses: actions/checkout@v4.1.1
+
+      - name: Install Dependencies
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build doxygen graphviz
```
Comment (byrnHDF) [inline .github/workflows/julia-cmake.yml:26]: Might want to use the doxygen action to specify version 1.9.6

Comment (byrnHDF) [reply .github/workflows/julia-cmake.yml:26]: - name: Install Dependencies
        uses: ssciwr/doxygen-install@v1
        with:
          version: "1.9.7"

File: .github/workflows/julia-auto.yml (hunk 1 above)
Comment (derobins) [reply .github/workflows/julia-auto.yml]: Fixed

**Review by hyoklee — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

**Comment by derobins** [comment]
Fixes #4539

**Pull request merged**